### ISSUE TITLE
Sometimes I can't move Chemistry book down with down arrow key

### ISSUE THREAD

**Issue by terracoda** [phetsims/friction#86]
I haven't figured out exactly when this happens, but sometimes after a breakaway I cannot move the Chemistry book down to be against the Physics book (visually)., but when I move the Chemistry book, the atoms behave as if they are pressed against the Physics book.

In the screen shot below, you see the focus highlight is against the Physics book, but the Chemistry book is not. Maybe that's the issue? Sometimes the model and focus highlight still think there are some atoms present in a row where they are all gone, causing a visual gap?

![screen shot 2018-08-07 at 10 42 46 am](https://user-images.githubusercontent.com/1902116/43778016-12672058-9a2f-11e8-8551-04e4705a5035.png)

**Comment by zepumph** [comment]
And of course, if you do run into it again and can report steps to reproduce, that would be great.

**Comment by terracoda** [comment]
@zepumph, this happened again for me a few times. Perhaps it is a Mac specific issue?

However, that said. Could it be possible be that sometimes it looks like all the atoms in the row have broken away visually, but in the model, the model still thinks one last atom is left in that row. That is what it seems like to me.

This happened in BASE in early designs. For a while the negative charge was off by one in the model, so the balloon could behave like it have negative charge, but visually it was showing as zero net charge.

**Comment by terracoda** [comment]
@zepumph, I do think it just might be what I described above, an invisible atom that is still there in the row. I think I just saw one appear upon evaporation when I got the big gap and continued to rub.

**Comment by zepumph** [comment]
@terracoda and I just spent a lot of time working on this, but we found could only reproduce it once (safari w/o VoiceOver on); I have yet to reproduce it on win10 chrome/firefox. 

We think that the problem has to do with the model/view thinking different things about whether or not there are any more atoms on a row. When this bug occurs, you can still "evaporate" the empty line, so this really is a visual bug.

@KatieWoe or @phet-steele  do you mind cranking at this to see if you can possibly come up with an easy way to reproduce it. Without that, it will be difficult to figure out how the bug is occurring.

**Comment by phet-steele** [comment]
@terracoda @zepumph I could really use a version to work with here.

**Comment by zepumph** [comment]
master!

**Comment by KatieWoe** [comment]
I just managed to recreate it with Mac OS 10.13 with voice over.  Now that I have an idea of how it happened I'll try to recreate it on other platforms.
Steps I used:
1. On Mac 10.13 Safari turn on voice over and keyboard nav
2. Use shift + wasd to move slowly shift the zoomed in book
3. Move the book back and forth as slowly as possible while still increasing temperature
4. Stop **as soon** as the layer disappears.  Note: This seems to occur mostly when the last destructible layer is disappearing
5. If you have stopped at the right moment, the visual will seem to show no atoms, but the box showing the control area will still be larger, as shown in above screenshot.
6. If you continue to move this glitched book back and forth the rest of the invisible layer will rub off and the sim will look normal

**Comment by KatieWoe** [comment]
One last check.  Made it happen on Win 10 Firefox.  This time it happened on the very first layer.
![image](https://user-images.githubusercontent.com/41024075/45244018-3ef60200-b2b3-11e8-8a0a-9a427d966f99.png)
@zepumph Not very easy, it takes a few tries to get right, but I hope this helps.

**Comment by terracoda** [comment]
This gap issue can even happen when focus is on the Chemistry book. Sorry, I do not know how to reproduce it.

**Comment by zepumph** [comment]
After some investigation, I found the issue to be in the logic of `FrictionModel.tryToEvaporate()`. Basically there are two conditional pieces in this method (I'm about to nerd out, just for fun; this isn't important for anyone to know). 

1. If you have any atoms left in current row that haven't already evaporated, randomly select one and evaporate it.
2. If all of the atoms are evaporated, point to the next row of evaporable atoms.

The bug was that the second conditional was before the first one, so in the worst case, you could evaporate the last atom in the row at the end of the method, and then find that the pointer to the current "evaporable row" was looking at a row with no visible atoms in it. The view of the atoms (and the focusHighlight) only updated during the second conditional. Thus, when you timed it just right, such that the last atom of a row was the last atom to evaporate during that session, then you got the space below.

@KatieWoe please see if you can confirm fixed on phettest.

**Comment by KatieWoe** [comment]
@zepumph unable to recreate on master after about 10 minutes of testing on Win 10 firefox.  As mentioned in my comments, this issue was sometimes rather tricky to recreate, but I think we can be reasonably confident that it is fixed.  Still, might be a good idea for @terracoda and @phet-steele to take a look as well.
I did note that there seemed to be some odd white flashes while testing the sim.

**Comment by zepumph** [comment]
Assigning to @phet-steele. Please close if this seems fixed.

**Comment by terracoda** [comment]
@phet-steele, I have not noticed this issue over the past two days of testing. I think this is a good sign even though I haven't been trying to re-create it. It used to happen quite frequently.

**Comment by phet-steele** [comment]
This was hard enough as it was to reproduce, so if no one has seen it than we can call it good.
